packages/language/src/document.js

```diff
@@ -0,0 +1,741 @@
+ *     detected from the text. This is only present if detections of this type
+ *     were found.
+ * @param {string[]} callback.annotation.entities.goods - Consumer good entities
+ *     detected from the text. This is only present if detections of this type
+ *     were found.
+ * @param {string[]} callback.annotation.entities.organizations - Organization
+ *     entities detected from the text. This is only present if detections of
+ *     this type were found.
+ * @param {string[]} callback.annotation.entities.other - Other entities
+ *     detected from the text. This is only present if detections of this type
+ *     were found.
+ * @param {string[]} callback.annotation.entities.people - People entities
+ *     detected from the text. This is only present if detections of this type
+ *     were found.
+ * @param {string[]} callback.annotation.entities.places - Place entities
+ *     detected from the text. This is only present if detections of this type
+ *     were found.
+ * @param {string[]} callback.annotation.entities.unknown - Unknown entities
+ *     detected from the text. This is only present if detections of this type
+ *     were found.
+ * @param {string[]} callback.annotation.sentences - Sentences detected from the
+ *     text.
+ * @param {object[]} callback.annotation.tokens - Parts of speech that were
+ *     detected from the text.
+ * @param {object[]} callback.annotation.tokens.text - The piece of text
+ *     analyzed.
+ * @param {object[]} callback.annotation.tokens.partOfSpeech - A description of
+ *     the part of speech (`Noun (common and propert)`,
+ *     `Verb (all tenses and modes)`, etc.).
+ * @param {object[]} callback.annotation.tokens.partOfSpeechTag - A short code
+ *     for the type of speech (`NOUN`, `VERB`, etc.).
+ * @param {object} callback.apiResponse - The full API response.
+ *
+ * @example
+ * document.annotate(function(err, annotation) {
+ *   if (err) {
+ *     // Error handling omitted.
+ *   }
+ *
+ *   // annotation = {
+ *   //   language: 'en',
+ *   //   sentiment: 100,
+ *   //   entities: {
+ *   //     organizations: [
+ *   //       'Google'
+ *   //     ],
+ *   //     places: [
+ *   //       'American'
+ *   //     ]
+ *   //   },
+ *   //   sentences: [
+ *   //     'Google is an American multinational technology company ' +
+ *   //     'specializing in Internet-related services and products.'
+ *   //   ],
+ *   //   tokens: [
+ *   //     {
+ *   //       text: 'Google',
+ *   //       partOfSpeech: 'Noun (common and proper)',
+ *   //       partOfSpeechTag: 'NOUN'
+ *   //     },
+ *   //     {
+ *   //       text: 'is',
+ *   //       partOfSpeech: 'Verb (all tenses and modes)',
+ *   //       partOfSpeechTag: 'VERB'
+ *   //     },
+ *   //     ...
+ *   //   ]
+ *   // }
+ * });
+ *
+ * //-
+ * // To request only certain annotation types, provide an options object.
+ * //-
+ * var options = {
+ *   entities: true,
+ *   sentiment: true
+ * };
```

Comment:
I think this is just dealing with how `annotate` works-- you can ask for 1, 2, or 3 annotations, so the options just map to the endpoint behavior.

> I don't think we have a similar mechanism anywhere else in our codebase.

Vision has similar behavior with `detect()` where you specify what features you want:
```js
vision.detect('image.jpg', { features: ['face', 'label'] }, function() {}) 
```

I considered an array of strings here, too, but it felt like too much boilerplate:
```js
doc.annotate('image.jpg', { annotations: ['syntax'] }, function() {})
```

Let me know if you think that's better, though.

> In this example it seems like it would be simpler to just say syntax: false and more predictable since all are enabled by default.

I generally think whitelists are better than blacklists, i.e., it's better for a user to be specific about what they want rather than finding out everything that exists, and choosing what they don't want.

I can see how this whole thing is weird, though, since if a user just looks at *our* API, it's like there is only one API method upstream, `annotate` and we just create some convenience methods around it. But that's not the case, since there are `detectEntities` and `detectSentiment` API endpoints upstream. The API must distinguish for a reason, so I'm not comfortable saying `detectSentiment()` is the same as `annotate({ sentiment: true })`. I can see the upstream API merging these two or making the relationship more clear in the future, but for now, we kind of have to follow their schematic, since we don't know what goes on behind the scenes after a request is made.

I can see our API getting an overhaul/simplification in the future when the API makes some moves. For now, though, I think this is the safest path forward.